intermediate_source/torchserve_with_ipex_2.rst

```diff
@@ -0,0 +1,471 @@
+
+Exercise
+^^^^^^^^
+Let's demonstrate channels last optimization. 
+
+.. code:: python
+
+    import torch
+
+    class Model(torch.nn.Module):
+        def __init__(self):
+            super(Model, self).__init__()
+            self.conv = torch.nn.Conv2d(16, 33, 3, stride=2)
+            self.relu = torch.nn.ReLU()
+
+        def forward(self, x):
+            x = self.conv(x)
+            x = self.relu(x)
+            return x 
+
+    model = Model()
+    model.eval()
+    data = torch.rand(20, 16, 50, 100)
+
+    import intel_extension_for_pytorch as ipex
+    ############################### code changes ###############################
+    ipex.disable_auto_channels_last() # omit this line for channels_last (default) 
+    ############################################################################
+    model = ipex.optimize(model)
+
+    with torch.no_grad():
+        model = torch.jit.trace(model, data)
+        model = torch.jit.freeze(model)
+
+We'll use `oneDNN verbose mode <https://oneapi-src.github.io/oneDNN/dev_guide_verbose.html>`_, a tool to help collect information at oneDNN graph level such as operator fusions, kernel execution time spent on executing oneDNN primitives. For more information, refer to the `oneDNN Documentation <https://oneapi-src.github.io/oneDNN/index.html>`_.
+
+.. figure:: /_static/img/torchserve-ipex-images-2/16.png
+   :width: 15%
+   :align: center
+   
+.. figure:: /_static/img/torchserve-ipex-images-2/17.png
+   :width: 100%
+   :align: center
+   
+Above is oneDNN verbose from channels first. We can verify that there are reorders from weight and data, then do computation, and finally reorder output back.   
+
+.. figure:: /_static/img/torchserve-ipex-images-2/18.png
+   :width: 80%
+   :align: center
+   
+Above is oneDNN verbose from channels last. We can verify that channels last memory format avoids unnecessary reorders.
+
+Intel® Extension for PyTorch* Integration into TorchServe
+========================================================= 
+Intel® Extension for PyTorch* has already been integrated into TorchServe to improve the performance out-of-box. :superscript:`2` When TorchServe creates single or multiple instances for inference, the launcher from Intel® Extension for PyTorch* is invoked automatically to set CPU affinity to these instances when launcher is enabled. :superscript:`3` This helps to make each instance use its assigned resources as high efficiently as possible, and minimize resource conflict among instances. For custom handler scripts, we recommend adding the *intel_extension_for_pytorch* package in.
+
+2. The feature has to be explicitly enabled by setting *ipex_enable=true* in  *config.properties*.
```

Comment:
should lines 383 and 384 be converted int one?

Reply:
Let's keep the two features (and lines 383, 384) separate to be explicit.